src/roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -73,33 +77,37 @@ def __init__(self, file_path: PathType, optical_series_name: Optional[str] = "Tw
                 raise ValueError("No acquisitions found in the .nwb file.")
             self._optical_series_name = a_names[0]
 
-        self.two_photon_series = self.nwbfile.acquisition[self._optical_series_name]
+        self.photon_series = self.nwbfile.acquisition[self._optical_series_name]
+        photon_series = dict(
+            OnePhotonSeries=OnePhotonSeries,
+            TwoPhotonSeries=TwoPhotonSeries,
+        )[self._optical_series_name]
         assert isinstance(
-            self.two_photon_series, TwoPhotonSeries
-        ), "The optical series must be of type pynwb.TwoPhotonSeries"
+            self.photon_series, photon_series
+        ), f"The optical series must be of type {photon_series.__name__}"
```

Comment:
change to is at least OnePhoton or TwoPhoton, the optical series name can be anything not necessarily a valid photon series type so need to change that.